src/Platforms/AbstractPlatform.php

```diff
@@ -4027,17 +4033,11 @@ private function columnToArray(Column $column): array
     {
         $name = $column->getQuotedName($this);
 
-        $columnData = array_merge($column->toArray(), [
+        return array_merge($column->toArray(), [
             'name' => $name,
             'version' => $column->hasPlatformOption('version') ? $column->getPlatformOption('version') : false,
             'comment' => $this->getColumnComment($column),
         ]);
-
-        if ($columnData['type'] instanceof Types\StringType && $columnData['length'] === null) {
-            $columnData['length'] = $this->getVarcharDefaultLength();
```

Comment:
This is already implemented in `getVarcharTypeDeclarationSQL()`: https://github.com/doctrine/dbal/blob/fade2e228b180bc44e4e7ef00bbfe0107d38f4a2/src/Platforms/AbstractPlatform.php#L208-L210

If this block is left as is, then it will be impossible to detect whether the column length was specified explicitly or the default value is used.